File: src/main/scala/org/deeplearning4j/scala/AudioVectorizer.scala
package org.deeplearning4j.scala

import org.canova.api.split.FileSplit
import java.io.File
import org.canova.sound.recordreader.WavFileRecordReader
import org.canova.common.RecordConverter

/**
 * Created by sogolmoshtaghi on 4/21/15.
 */
object AudioVectorizer {

  def main(args: Array[String]) {
    val recordReader = new WavFileRecordReader(true)
    val filePath = System.getProperty("user.home") + "/dsr/data/mlsp_contest_dataset/essential_data/src_wavs"

    recordReader.initialize(new FileSplit(new File(filePath)))

    while (recordReader.hasNext()) {
      val dataset = recordReader.next()
      val arr = RecordConverter.toArray(dataset);
      println(arr)
    }

  }
}

File: src/main/java/datapipelines/AudioVectorizer.java
package datapipelines;

import org.canova.api.records.reader.RecordReader;
import org.canova.api.split.FileSplit;
import org.canova.api.writable.Writable;
import org.canova.common.RecordConverter;
import org.canova.sound.recordreader.WavFileRecordReader;
import org.deeplearning4j.datasets.canova.RecordReaderDataSetIterator;
import org.deeplearning4j.datasets.iterator.DataSetIterator;
import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.dataset.api.DataSet;

import java.io.File;
import java.util.Collection;


public class AudioVectorizer {
    public static void main( String[] args )  throws Exception {

        String filepath = System.getProperty("user.home")
                +"/dsr/data/mlsp_contest_dataset/essential_data/src_wavs";

        RecordReader wavRecordReader = new WavFileRecordReader(true);
        wavRecordReader.initialize(new FileSplit(new File(filepath)));

        DataSetIterator iter = new RecordReaderDataSetIterator(wavRecordReader);

        while(iter.hasNext()) {
           DataSet next = iter.next();
           //network.fit(next);
       }

    }

}